src/utils/auth/awsSignature.ts

```diff
@@ -0,0 +1,17 @@
+import aws4 = require('aws4');
+import got = require('got');
+
+export function awsSignature(authorization: string): got.BeforeRequestHook<got.GotBodyOptions<null>> {
+    const [ , accessKeyId, secretAccessKey ] = authorization.split(" ");
```

Comment:
Thank you so much for your patience. I split the header by any number of whitespace like you suggested.

I was considering using the already split result, but I was hesitant because of the names of the variables (`user` and `pass`/`args`) and didn't want to change too much of the existing code. Maybe it will make more sense to make it more general if there are even more additional authentication types?